File: SDKs/github.com/ghes-3.0/1.1.4/python/src/sdk/models/operations/activity_delete_thread_subscription.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class ActivityDeleteThreadSubscriptionPathParams:
    thread_id: int = field(default=None, metadata={'path_param': { 'field_name': 'thread_id', 'style': 'simple', 'explode': False }})
    

@dataclass
class ActivityDeleteThreadSubscriptionRequest:
    path_params: ActivityDeleteThreadSubscriptionPathParams = field(default=None)
    

@dataclass
class ActivityDeleteThreadSubscriptionResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    basic_error: Optional[shared.BasicError] = field(default=None)
    

File: SDKs/github.com/ghes-3.0/1.1.4/typescript/src/sdk/models/operations/activitydeletethreadsubscription.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class ActivityDeleteThreadSubscriptionPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=thread_id" })
  threadId: number;
}


export class ActivityDeleteThreadSubscriptionRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: ActivityDeleteThreadSubscriptionPathParams;
}


export class ActivityDeleteThreadSubscriptionResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  basicError?: shared.BasicError;
}
